DigitalLearningSolutions.Data/DataServices/CourseDataService.cs

```diff
@@ -593,7 +593,8 @@ public IEnumerable<DelegateCourseInfo> GetDelegateCourseInfosForCourse(int custo
                                             cap.Active = 1)))
                         AND da.CentreID = @centreId
                         AND pr.CustomisationID = @customisationId
-                        AND ap.DefaultContentTypeID <> 4",
+                        AND ap.DefaultContentTypeID <> 4
+                        AND TRY_CAST(COALESCE(ucd.Email, u.PrimaryEmail) AS UNIQUEIDENTIFIER) IS NULL",

```

Comment:
I think it would be easier to achieve this using LINQ on the result set returned rather than in the SQL query? This would avoid the need to introduce an additional excludeGuid parameter for the service method too.

Reply:
@kevwhitt-hee I wonder if we should have some sort of sql variables file for magic numbers like the 4 here also? Just to make it more obvious exactly what the meaning is here?

Reply:
This is a good idea. Probably not worth holding this ticket up for, though, since we didn't introduce it here. I'll add a maintenance/tech debt ticket to tidy these up.